### DISCUSSION TITLE
Does it make sense to get rid of the ResultEntity UUIDS?

### DISCUSSION THREAD

**Discussion by t-ober** [ie3-institute/simona#415] [General]
Since the TimeStamp (after making sure it includes seconds: #413 ) should uniquely identify every result it might not be necessary to include them.

**Comment by sebastian-peter** [comment]
I think so too (at least time stamps **in combination with** input model UUID). Same for primary data.

**Comment by sebastian-peter** [comment]
There could be difficulties identifying (and deleting) duplicate rows, once they exist. But having row UUIDs do not help either if, let's say you have two rows with identical datetimes and input models, but different power values.

With SQL, one can easily define a unique index on uuid and input model, preventing duplicate entries from being entered.

**Reply by t-ober** [reply]
As far as I can tell this would also be an error within the simulation since there should never be two different results for a participant within the same second. If necessary one could put up a safeguard in the `ResultEventListener` that remembers the last result for every participant. And either overwrites the last value and/or informs about two result having been produced for the same result.

**Reply by sebastian-peter** [reply]
Yes, we could do that, but I wouldn't argue that that's even necessary. As you said, results should be unambiguous for input model + datetime and I think it's fine if we just assume that they are.

**Reply by sebastian-peter** [reply]
Btw, when we provide SQL documentary, we should include recommendations for `CREATE TABLE`-statements, so that we also include index creation. -> https://github.com/ie3-institute/PowerSystemDataModel/issues/526#issuecomment-1327135946

**Comment by sebastian-peter** [comment]
Created issue here: https://github.com/ie3-institute/PowerSystemDataModel/issues/826
